kong/plugins/prometheus/exporter.lua

```diff
@@ -29,9 +29,15 @@ local function init()
   prometheus = require("kong.plugins.prometheus.prometheus").init(shm, "kong_")
 
   -- global metrics
-  metrics.connections = prometheus:gauge("nginx_http_current_connections",
-                                         "Number of HTTP connections",
-                                         {"state"})
+  if ngx.config.subsystem == "http" then
+    metrics.connections = prometheus:gauge("nginx_http_current_connections",
+      "Number of HTTP connections",
+      {"state"})
+  else
+    metrics.connections = prometheus:gauge("nginx_stream_current_connections",
+      "Number of Stream connections",
+      {"state"})
```

Comment:
The naming here is not idiomatic here based on the Prometheus metrics naming guide: https://prometheus.io/docs/instrumenting/writing_clientlibs/

This is not a problem because of lower usage of non-http based use-cases of Kong.
We will need to change this (breaking change) at some point in future though.

This is small enough right now that we shouldn't be worrying about it. I'm leaving this comment for knowledge share. No changes required in the patch itself.